### DISCUSSION TITLE
Language translation support

### DISCUSSION THREAD

**Discussion by empanadadexoubas** [facileManager/facileManager#691] [General]
Hi all,

I've seen that titles and texts are directly written in php files, instead of using a properties file or a database table.
It would be possible to modify that to add language support?

Thank you.

**Comment by WillyXJ** [comment]
Hi,

I think this should be possible by using gettext.  fM wasn't built with it because I didn't think the project would become so large and popular. :)  Thanks for the request and I'll try implementing it in a future version.

Thanks.

**Comment by WillyXJ** [comment]
Language support will be included in 2.0-beta1 and later; however, English will be the only supported language until the POT files get translated by contributors.

**Comment by Vringe** [comment]
I would like to add a translation. How can this be done?
I have created a new folder in /fm-modules/facileManager/languages/de_CH/LC_MESSAGES and uploaded the po und mo file.

Thank you for creating this awesome project! :)

**Comment by WillyXJ** [comment]
Hi Vringe,

Thank you for your interest in translating.  You will need to give me a pull request through Git with the files so I can merge them into the master branch, test, and package up for the public.

**Comment by Vringe** [comment]
Hi WillyXJ,

Sorry, I'm not familiar with git. Could you please give me the commands to
make a pull request?
I have already done a "git clone" and added the translation files.

Zitat von WillyXJ notifications@github.com:

> Hi Vringe,
> 
>   Thank you for your interest in translating. You will need to give me a
> pull request through Git with the files so I can merge them into the
> master branch, test, and package up for the public.
> 
>   —
> Reply to this email directly or view it on GitHub[1].
>       

## Links:

[1] https://github.com/WillyXJ/facileManager/issues/106#issuecomment-85127116

**Comment by WillyXJ** [comment]
Hi @Vringe - you should be able to do it through github.com or the Mac/Windows client which makes it easy - https://help.github.com/articles/using-pull-requests/

**Comment by Vringe** [comment]
Hi WillyXJ,

I think I made it ;-)

> Hi @Vringe https://github.com/Vringe  - you should be able to do it through github.com or the Mac/Windows client which makes it easy - https://help.github.com/articles/using-pull-requests/
> 
> —

**Comment by WillyXJ** [comment]
We have working POT files for other translators to use.

**Comment by empanadadexoubas** [comment]
Thank you for your effort!

De: WillyXJ [mailto:notifications@github.com] 
Enviado el: viernes, 04 de diciembre de 2015 0:39
Para: WillyXJ/facileManager facileManager@noreply.github.com
CC: empanadadexoubas daniel.rua@altia.es
Asunto: Re: [facileManager] language support (#106)

We have working POT files for other translators to use.

**Comment by adrianreed** [comment]
Hello, I'm not really familiar with Git, but as soon as I find out how to, I'll happily contribute with Spanish translations.
I've been looking for a DNS manager for a while, and the past couple days I've been testing the fmDNS module on VMs to simulate my two DNS servers. So far, apart from a couple troubles during installation, everything's going great. Thank you very much!

**Comment by WillyXJ** [comment]
Sounds good, @adrianvergara.  Thank you for your support!

**Comment by Pierre-CHABREDIER** [comment]
I have created a french translation with poedit for facileManager and fmDNS, but I dont know how to upload it to You (dev team).

If someone can send me a brief explanation of how to pull it with git ?

Thank You in advance.

Regards.

**Comment by WillyXJ** [comment]
@Pierre-CHABREDIER - Thanks for contributing! See if this document helps you - https://help.github.com/articles/creating-a-pull-request/

Also, could you include translations for fmFirewall and fmSQLPass?

**Comment by Pierre-CHABREDIER** [comment]
Ok, I've tried to understand what is a branch etc... but in fact I don't understand anything.

So here you can find french translations for facileManager and fmDNS :
[fm-modules.zip](https://github.com/WillyXJ/facileManager/files/226463/fm-modules.zip)

Also, as I don't have fmFirewall and fmSQLPass, I don't have translations for these modules. But if I have 1 or 2 free hours, I will try to translate them also.

**Comment by WillyXJ** [comment]
I am looking to move translations to either [Weblate](https://weblate.org/) or [Crowdin](https://crowdin.com/) which I hope will make it easier on contributors.
